src/plugins/explore/public/application/register_tabs.ts

```
@@ -89,8 +115,20 @@ export const registerBuiltInTabs = (tabRegistry: TabRegistryService) => {
  * This is the main entry point for tab registration
  */
 export const registerTabs = (services: ExploreServices) => {
-  // Register built-in tabs
-  registerBuiltInTabs(services.tabRegistry);
+  // Get the current flavor from the URL
+  const path = window.location.pathname;
+  let flavor: ExploreFlavor | undefined;
+
+  if (path.includes(`/${ExploreFlavor.Logs}`)) {
+    flavor = ExploreFlavor.Logs;
+  } else if (path.includes(`/${ExploreFlavor.Traces}`)) {
+    flavor = ExploreFlavor.Traces;
+  } else if (path.includes(`/${ExploreFlavor.Metrics}`)) {
+    flavor = ExploreFlavor.Metrics;
+  }
```

lets throw an error if its not one of them